### PR TITLE
Add pbkdf2 encode/decode for credential sensitive field

### LINKED ISSUE #558: Add AES encode/decode for credential sensitive field

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #558

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]
@WangLiNaruto we are only encoding the value and storing. How/where are we handling the decoding?

**Comment by WangLiNaruto** [comment]
> @WangLiNaruto we are only encoding the value and storing. How/where are we handling the decoding?

@swaroopar Yes, we just encoded sensitive fields and stored

**Comment by WangLiNaruto** [comment]
> @WangLiNaruto we are only encoding the value and storing. How/where are we handling the decoding?

File: modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java [hunk 1]
```diff
@@ -174,7 +179,15 @@ public void addCredential(CreateCredential createCredential) {
...
                             createCredential.getCsp(), createCredential.getType()));
         }
-        createCredential(credential);
+    }
+
+    private void encodeSensitiveVariables(CreateCredential createCredential) {
+        List<CredentialVariable> variables = createCredential.getVariables();
+        variables.stream().forEach(variable -> {
+            if (!Objects.isNull(variable) && variable.getIsSensitive()) {
+                variable.setValue(passwordEncoder.encode(variable.getValue()));
```
Comment (baixinsui) [inline modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java:188]: When variable.getValue() is null, encode (null) will throw NullPointerException，We should add a judgment condition Objects.isNull(variable.getValue()) ;

**Pull request closed without merging**